frontend/Auth/FinishResetPassword.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from "react";
+import { Button, ButtonKind } from "../Styleguide/Button";
+import { FormField } from "../Styleguide/FormField";
+import { FormFieldLabel } from "../Styleguide/FormFieldLabel";
+import { Input } from "../Styleguide/Input";
+import { RouterProps } from "react-router";
+import { useMutation, useQuery } from "react-query";
+import { flaxFetch } from "../Utils/flaxFetch";
+import { unary } from "lodash-es";
+
+export function FinishResetPassword(props: RouterProps) {
+  const [finishResetPasswordFormData, setFinishResetPasswordFormData] =
+    useState<FinishResetPasswordFormData>({
+      password: "",
+    });
+
+  const paramString = props.history.location.search;
+  let searchParams = new URLSearchParams(paramString);
+  const token = searchParams.get("jwt") || "MISSING TOKEN";
+
+  const queryFunctionHelper = (options) => {
+    const token = options.queryKey[0];
+    return flaxFetch<TokenValidationResponse>(
+      `/validate-token/${token}?tokenType=passwordReset`,
+      {
+        method: "GET",
+        signal: options.signal,
+      }
+    );
+  };
+
+  const tokenValidationResponse = useQuery<TokenValidationResponse>(
+    token,
+    queryFunctionHelper
+  ).data;
+
+  const userEmail = tokenValidationResponse?.email;
+
+  const submitMutation = useMutation<
+    void,
+    Error,
+    React.FormEvent<HTMLFormElement>
+  >((evt) => {
+    evt.preventDefault();
+    const ac = new AbortController();
+    let requestPromise = flaxFetch<void>(`/send-reset-password-email`, {
```

Comment:
It would maybe be nicer to just skip making the request to the backend if there is no token (when it's `"MISSING TOKEN"`). Something like this:

```ts
if (token) {
  return flaxFetch(...)
} else {
  return Promise.resolve({
    tokenIsValid: false,
    tokenIsExpired: false,
    email: null
  })
}
```

Reply:
Additionally `<void>` is not the correct response body for this request. flaxFetch should allow you to just provide no generic (`flaxFetch()`). If it doesn't, we should fix that.

```suggestion
    let requestPromise = flaxFetch(`/send-reset-password-email`, {
```

Reply:
Pair programmed to resolve.